tools/analyzer_plugin/lib/src/diagnostic/invalid_child.dart

```diff
@@ -52,15 +51,19 @@ Future<void> validateReactChildType(DartType type, TypeSystem typeSystem,
   if (type.isReactElement) return;
   if (type.isDartCoreString) return;
   // isAssignableTo to handle num, int, and double
-  if (typeSystem.isAssignableTo(typeSystem.typeProvider.numType, type)) return;
+  if (typeSystem.isAssignableTo(typeProvider.numType, type)) return;
   if (type.isDartCoreNull) return;
   if (type.isDartCoreBool) return;
   // If the children are in an iterable, validate its type argument.
   // To check for an iterable, type-check against `iterableDynamicType` and not
   // `iterableType` since the latter has an uninstantiated type argument of `E`.
-  if (typeSystem.isSubtypeOf(type, typeSystem.typeProvider.iterableDynamicType)) {
-    var typeArg = typeSystem.mostSpecificTypeArgument(type, typeSystem.typeProvider.iterableDynamicType);
```

Comment:
Apparently this API was only available on a src import, so I implemented a grosser workaround using public APIs.